lib/streamlit/elements/exception.py

```diff
@@ -205,4 +217,15 @@ def _get_stack_trace_str_list(exception, strip_streamlit_stack_entries=False):
 
     stack_trace_str_list = [item.strip() for item in stack_trace_str_list]
 
+    if isinstance(exception, UncaughtAppException):
```

Comment:
Hm, is `exception` totally unused here? I feel like it'd be more natural to get the stacktrace directly from `exception` rather than `traceback.format_exc()` if that's possible